factcast-store-pgsql/src/main/java/org/factcast/store/pgsql/registry/transformation/TransformationCacheCompactor.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright © 2017-2020 factcast.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.factcast.store.pgsql.registry.transformation;
+
+import org.factcast.store.pgsql.registry.transformation.cache.TransformationCache;
+import org.joda.time.DateTime;
+import org.springframework.scheduling.annotation.Scheduled;
+
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.Value;
+import lombok.extern.slf4j.Slf4j;
+import net.javacrumbs.shedlock.core.SchedulerLock;
+
+@RequiredArgsConstructor
+@Slf4j
+@Value
+public class TransformationCacheCompactor {
+    @NonNull
+    final TransformationCache cache;
+
+    final int days;
+
+    @Scheduled(
+            cron = "${factcast.store.pgsql.transformationCacheCompactCron:0 0 0 * * *}")
+    @SchedulerLock(name = "transformationCacheCompact", lockAtMostFor = 1000 * 60 * 60)
+    public void compact() {
+
+        // TODO at timing metrics
```

Comment:
Its done inside `compact`. Should I move it over here?

Reply:
then it would also include the inmem metrics. wouldn't that be better?

Reply:
Before these changes the compact of the InMemImpl was a `cache.clear();` so I thought it wasnt necessary 😂
Sure move it :)